Week1/homework/app.js

```diff
@@ -3,9 +3,162 @@
+    },
+    animal_farm: {
+      title: 'Animal Farm',
+      author: 'George Orwell',
+      language: 'English',
+      length: 112,
+      info:
+        'Mr Jones of Manor Farm is so lazy and drunken that one day he forgets to feed his livestock. The ensuing rebellion under the leadership of the pigs Napoleon and Snowball leads to the animals taking over the farm. Vowing to eliminate the terrible inequities of the farmyard, the renamed Animal Farm is organised to benefit all who walk on four legs. But as time passes, the ideals of the rebellion are corrupted, then forgotten. And something new and unexpected emerges. . .'
+    },
+    a_tale_of_two_cities: {
+      title: 'A Tale of Two Cities',
+      author: 'Charles Dickens',
+      language: 'English',
+      length: 544,
+      info:
+        'After eighteen years as a political prisoner in the Bastille the aging Dr Manette is finally released and reunited with his daughter in England. There two very different men, Charles Darnay, an exiled French aristocrat, and Sydney Carton, a disreputable but brilliant English lawyer, become enmeshed through their love for Lucie Manette. From the tranquil lanes of London, they are all drawn against their will to the vengeful, bloodstained streets of Paris at the height of the Reign of Terror and soon fall under the lethal shadow of La Guillotine.'
+    },
+    the_alchemist: {
+      title: 'The Alchemist',
+      author: 'Paulo Coelho',
+      language: 'English',
+      length: 197,
+      info:
+        'Paulo Coelhos masterpiece tells the mystical story of Santiago, an Andalusian shepherd boy who yearns to travel in search of a worldly treasure.His quest will lead him to riches far different—and far more satisfying—than he ever imagined.Santiagos journey teaches us about the essential wisdom of listening to our hearts, of recognizing opportunity and learning to read the omens strewn along lifes path, and, most importantly, to follow our dreams.'
+    }
+  };
+  const BookCovers = {
+    moby_dick: 'img/moby_dick.jpg',
+    the_adventures_of_huckleberry_finn: 'img/the_adventures_of_huckleberry_finn.jpg',
+    alices_adventures_in_wonderland: 'img/alices_adventures_in_wonderland.jpg',
+    charlie_and_the_chocolate_factory: 'img/charlie_and_the_chocolate_factory.jpg',
+    wuthering_heights: 'img/wuthering_heights.jpg',
+    war_and_peace: 'img/war_and_peace.jpg',
+    anna_karenina: 'img/anna_karenina.jpg',
+    animal_farm: 'img/animal_farm.jpg',
+    a_tale_of_two_cities: 'img/a_tale_of_two_cities.jpg',
+    the_alchemist: 'img/the_alchemist.jpg'
+  };
+
+  function generateBookCards(container) {
+    for (let i = 0; i < bookTitles.length + 1; i++) {
```

Comment:
You need not loop till `array length + 1` as the last element created would have its `id = undefined` i.e bookTitles is defined only from indices 0 to 9. `bookTitles[10] = undefined` 🙂

Reply:
Alternatively, you can make use of the `for...of` loop as you are using in the below functions.

Reply:
I realize that that issue after your review and fixed it